### PR TITLE
8290899: java/lang/String/StringRepeat.java test requests too much heap on windows x86

### PR DESCRIPTION
**Pull request by mrserb**
The `java/lang/String/StringRepeat.java` test was updated twice after integration:

https://bugs.openjdk.org/browse/JDK-8221400 - the xmx4g was replaced by the xmx2g
https://bugs.openjdk.org/browse/JDK-8265421 - the "os.maxMemory >= 2G" was added

Unfortunately, this test still may fail on Windows x86 due to: `Could not reserve enough space for xxx object heap.`

This is a request to exclude it on win x86 since that JDK cannot allocate 2g.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8290899](https://bugs.openjdk.org/browse/JDK-8290899): java/lang/String/StringRepeat.java test requests too much heap on windows x86

### Reviewers
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)
 * [Paul Hohensee](https://openjdk.org/census#phh) (@phohensee - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11639/head:pull/11639` \
`$ git checkout pull/11639`

Update a local copy of the PR: \
`$ git checkout pull/11639` \
`$ git pull https://git.openjdk.org/jdk pull/11639/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11639`

View PR using the GUI difftool: \
`$ git pr show -t 11639`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11639.diff">https://git.openjdk.org/jdk/pull/11639.diff</a>

</details>

### CODE REVIEW

**Comment by jaikiran** [comment]
Hello Sergey, the linked JBS issues note that this test takes at most 1400M. Should there be some investigation into why it's failing even with 2GB allowance?

**Comment by mrserb** [comment]
> Hello Sergey, the linked JBS issues note that this test takes at most 1400M. Should there be some investigation into why it's failing even with 2GB allowance?

On windows x86 java fails even before the start of the test, it is just impossible to allocate 2g of memory.
```
jdk/bin/java -mx2g --version
Error occurred during initialization of VM
Could not reserve enough space for 2097152KB object heap
```
The maximum it can allocate is around `-mx1650m`, but using that caused the test to fail from time to time.

**Review by jaikiran — APPROVED** [review]
Thank you for that detail. The change looks fine to me then. I had to look if multiple `@requires` are supported in a test definition by jtreg and apparently it does support it:

https://openjdk.org/jtreg/tag-spec.html
> The @requires tag may be used multiple times in a given test. If it is used more than once, the expressions in the individual instances will be enclosed in parentheses and combined with '&'.

Please wait for another review before integrating, since I don't have previous experience with this test.

**Comment by AlanBateman** [comment]
What you have looks okay if only Windows 32-bit is skipped but it might be simpler to just require sun.arch.data.model == "64" so that it only runs on 64-bit systems.

**Comment by mrserb** [comment]
> What you have looks okay if only Windows 32-bit is skipped but it might be simpler to just require sun.arch.data.model == "64" so that it only runs on 64-bit systems.

Right, but one of the fix for this test explicitly added support for x86
https://bugs.openjdk.org/browse/JDK-8221400
And on Linux x86 this test works fine.

**Comment by mrserb** [comment]
If there are no objections or any other comments I'll integrate the fix.

**Comment by phohensee** [comment]
This is a (very old) known problem with 32-bit Windows. Address space randomization (libraries allocated at widely dispersed locations within the 4g 32-bit virtual address space) means there's a limit to the size of contiguous virtual address space mappable to the Java heap.

**Review by phohensee — APPROVED** [review]
Lgtm.

**Pull request closed without merging**
